pkg/util/iptables/iptables.go

```diff
@@ -575,3 +576,252 @@ func IsNotFoundError(err error) bool {
 	}
 	return false
 }
+
+type RuleOption struct {
+	Arg     string
+	Negated bool
+}
+
+type Rule struct {
+	Modules []string
+	Jump    Chain
+	Comment string
+	Options map[string]RuleOption
+}
+
+// Split a line into space-separated arguments, handling simple quoting
+func splitLine(line string) ([]string, error) {
+	items := strings.Split(line, " ")
+	args := make([]string, 0, len(items))
+	for i := 0; i < len(items); i++ {
+		if items[i] == "" {
+			continue
+		}
+
+		if !strings.HasPrefix(items[i], "\"") {
+			args = append(args, items[i])
+			continue
+		}
+
+		// combine until we find a closing quote
+		var arg string
+		items[i] = items[i][1:]
+		for j := i; j < len(items); j++ {
+			if len(arg) > 0 {
+				arg += " "
+			}
+			arg += items[j]
+			if strings.HasSuffix(arg, "\"") {
+				args = append(args, arg[:len(arg)-1])
+				break
+			}
+			i++
+		}
+		if i == len(items) {
+			return nil, fmt.Errorf("invalid iptables rule (unterminated quote): %v", line)
+		}
+	}
+	return args, nil
+}
+
+func addRuleOption(rule *Rule, option string, ro RuleOption) error {
+	if rule.Options == nil {
+		rule.Options = make(map[string]RuleOption)
+	}
+	if _, ok := rule.Options[option]; ok {
+		return fmt.Errorf("invalid iptables rule (multiple %q args)", option)
+	}
+	rule.Options[option] = ro
+	return nil
+}
+
+var optionAttrs = map[string]struct {
```

Comment:
Maybe we can force the caller to pass in a list of flags it uses, so the list becomes site-specific, rather than global? The list below is what kube-proxy needs, but is not generic.